recipes-images/images/opentrons-ot3-image.bb

```diff
@@ -53,10 +37,178 @@ IMAGE_INSTALL += " \
     python3 python3-misc python3-modules \
  "
 
-do_image_zip() {
+# Prefix to the resulting deployable tarball name
+export IMAGE_BASENAME = "opentrons-ot3-image"
+MACHINE_NAME ?= "${MACHINE}"
+IMAGE_NAME = "${MACHINE_NAME}_${IMAGE_BASENAME}"
+SYSTEMFS_DIR = "${WORKDIR}/systemfs"
+USERFS_DIR = "${WORKDIR}/userfs"
+SYSTEMFS_OUTPUT = "${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.systemfs.ext4"
+USERFS_OUTPUT = "${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.userfs.ext4"
+
+# add the rootfs version to the welcome banner
+add_rootfs_version() {
+    printf "${DISTRO_NAME} ${DISTRO_VERSION} (${DISTRO_CODENAME}) \\\n \\\l\n" > ${IMAGE_ROOTFS}/etc/issue
+    printf "${DISTRO_NAME} ${DISTRO_VERSION} (${DISTRO_CODENAME}) %%h\n" > ${IMAGE_ROOTFS}/etc/issue.net
+    printf "${IMAGE_NAME}\n\n" >> ${IMAGE_ROOTFS}/etc/issue
+    printf "${IMAGE_NAME}\n\n" >> ${IMAGE_ROOTFS}/etc/issue.net
+}
+ROOTFS_POSTPROCESS_COMMAND += "add_rootfs_version;"
+
+# split the rootfs into systemfs and userfs
+fakeroot do_create_filesystem_trees() {
+    # this will create the systemfs tree
+    rsync -a ${IMAGE_ROOTFS}/ ${SYSTEMFS_DIR}
+
+    # create the userfs tree
+    rsync -a ${IMAGE_ROOTFS}/home ${USERFS_DIR}/
+    rsync -a ${IMAGE_ROOTFS}/var ${USERFS_DIR}/
+}
+IMAGE_POSTPROCESS_COMMAND += "do_create_filesystem_trees;"
+
+# create the ext4 filesystem
+do_create_filesystem() {
+    # get size of the filesystem trees
+    SYSTEMFS_SIZE=$(du -Lbks ${SYSTEMFS_DIR} | cut -f1)
+    USERFS_SIZE=$(du -Lbks ${USERFS_DIR} | cut -f1)
+ 
+    # create sparse file a bit larger than source dir
+    dd if=/dev/zero of=${SYSTEMFS_OUTPUT} seek=${SYSTEMFS_SIZE}w bs=1024 count=0
+    mkfs.ext4 -F ${SYSTEMFS_OUTPUT} -d ${SYSTEMFS_DIR}
+
+    dd if=/dev/zero of=${USERFS_OUTPUT} seek=${USERFS_SIZE}b bs=1024 count=0
+    mkfs.ext4 -F ${USERFS_OUTPUT} -d ${USERFS_DIR}
+}
+IMAGE_POSTPROCESS_COMMAND += "do_create_filesystem;"
+
+# create the tezi image.json
+python do_create_tezi_manifest(){
+    import os
+    import json
+    tezi_manifest_path = "%s/image-%s.json" % (d.getVar('DEPLOY_DIR_IMAGE'), d.getVar('IMAGE_BASENAME'))
+    tezi_ot3_manifest_path = "%s/image.json" % (d.getVar('DEPLOY_DIR_IMAGE'))
+    tezi_manifest = {}
+
+    # define the ot3  partitions
+    ot3_partitions = [{
```

Comment:
This doesn't quite look right. We want
- boot, FATFS, 48 meg
- RFS1, holding systemfs, fixed size (here, 3072M; would be good to cut this down if we can)
- RFS2, holding systemfs, same size as RFS1
- VARFS or whatever, holding userfs, remaining space on the storage

Right now, we've got bootfs and RFS1, but the volume labeled REFS2 has the userfs, and there's no dedicated partition for it.

Having two dedicated rootfs partitions allows us to update the partition we're not booted from.